### PR TITLE
Fix documentation rendering warnings

### PR DESCRIPTION
**Pull request by vsbogd**

### CODE REVIEW

**Comment by vsbogd** [comment]
@luketpeterson do we have a brief explanation how to build documentation for C API somewhere in repo? I cannot find it.

**Comment by luketpeterson** [comment]
> @luketpeterson do we have a brief explanation how to build documentation for C API somewhere in repo? I cannot find it.

I just added it here: https://github.com/trueagi-io/hyperon-experimental/pull/739

File: lib/src/space/mod.rs [hunk 1]
```diff
@@ -207,10 +207,10 @@ pub trait Space: std::fmt::Debug + std::fmt::Display {
         None
     }
 
-    /// Returns an &dyn [Any] for spaces where this is possible
+    /// Returns an &dyn [std::any::Any] for spaces where this is possible
```
Comment (luketpeterson) [inline lib/src/space/mod.rs:210]:
```suggestion
    /// Returns an `&dyn `[Any](std::any::Any) for spaces where this is possible
```

File: lib/src/space/mod.rs [hunk 2]
```diff
@@ -274,7 +274,7 @@ pub trait SpaceMut: Space {
     fn replace(&mut self, from: &Atom, to: Atom) -> bool;
 
     /// Turn a &dyn SpaceMut into an &dyn Space.  Obsolete when Trait Upcasting is stabilized.
-    /// https://github.com/rust-lang/rust/issues/65991  Any month now.
+    /// [Rust issue #65991](https://github.com/rust-lang/rust/issues/65991)  Any month now.
```
Comment (luketpeterson) [inline lib/src/space/mod.rs:277]: Looks like this language feature will be integrated into stable Rust in <2 months.  This missing feature was a huge obstacle for me when i was learning Rust in 2019.  But I've learned to work around it so I barely notice anymore.

**Review by luketpeterson — APPROVED** [review]

**Comment by vsbogd** [comment]
@luketpeterson thanks !

**Pull request merged**